qiskit/transpiler/oracle_compiler/types.py

```diff
@@ -0,0 +1,20 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Oracle function types."""
+
+from typing import NewType
+
+Bit = NewType('Bit', bool)
```

Comment:
I find it a bit confusing to have another Bit type additional to `qiskit.circuit.bit.Bit`, I probably don't fully understand this: why is the `Bit` type needed instead of just using `bool`? Is `Bit` the best name, if that is already used to define classical and quantum bits? 🤔

Reply:
> why is the `Bit` type needed instead of just using `bool`? 
The interoperability between python `bool` and `int` (to be added in the future) is ... complicated. At the end, we are going to need custom types anyway, since there is no way to as python for a 2-bit integer.

> Is `Bit` the best name?
Suggestions? `Int1`?

Reply:
The classical `qiskit.circuit.bit.Bit` do not work with python as bools. A nice thing of this solution is that we can check that the synthesis is working well because we can run the python function and check the result is the same as the synthesis execution.

Reply:
If the goal is to merge them I think its fine to have them called the same. But if you are planning on `IntN` for several `N` then `Int1` seems like a good solution to me too.

Reply:
Maybe @ajavadia have an opinion about it?

Reply:
yeah I think `Int1` is good.
https://llvm.org/docs/LangRef.html#examples

Reply:
Bit -> Int1 in 8c4dce8e184566e9012c2a6e840bfc1943aea567